### ISSUE TITLE
Have a flexible `quarto_get_meta()`

### ISSUE THREAD

**Issue by olivroy** [quarto-dev/quarto-r#182]
to be able to program around some things in the metadata from R

accessing the yaml metadata is currently difficult

**Comment by cderv** [comment]
Can you give some use case and example of usage on how this would work ? 

Quarto does not work as R Markdown and all the meta that will be used for Pandoc rendering won't be accessible on knitting time IMO. 

So I want clarify what is expected here. 

Let's remind that Quarto has some way to read the meta : 
* https://quarto.org/docs/authoring/variables.html#meta

and an extensions mechanism through Lua Filters to compute on the Meta 
* https://quarto.org/docs/extensions/filters.html

Unless we are not talking on the same metadata. 

Thanks

**Comment by olivroy** [comment]
In short, I'd like something that works 2 ways: (i.e. you set it in one place, and can use it as an r variable + quarto metadata.


I would just want to be able to use it as an R variable..

The meta shortcode is useful to print things, but not to manipulate it. I don'y know anything about lua, so I'd have to dig into it to find out more..

Let's say I want to read the date.


````qmd
---
title: "A title"
date: 2024-03-22
---


```{r}
month <- lubridate(ym(get_meta("date")
```

The month is `r month`

````

Basically, I would like to be able to set something once (either as R code OR in yaml meta) and let quarto AND R use it.

Alternatively, if the following worked (I think it did in rmarkdown.


````qmd
---
title: "A title"
date: 2024-03-22
---


```{r}
date <- "2024-03-22"
month <- lubridate::month(date)
```

---
date: !expr date
---

`r month`

````
````

**Comment by cderv** [comment]
> Basically, I would like to be able to set something once (either as R code OR in yaml meta) and let quarto AND R use it.

This would be a feature request to Quarto itself so that metadata are accessible using objects inside computation cells. It is not just a Quarto R package feature. This 📦 is only a wrapper around quarto CLI, which additional helpers for R users. 

This is much more a feature that would need to work for all engine in Quarto (Jupyter, Knitr, Julia, ...)

You can probably open a discussion there about this, if none exists already

> Alternatively, if the following worked (I think it did in rmarkdown.

This is about 
* https://github.com/quarto-dev/quarto-r/issues/137

It works today, you just need to drop the `!expr` which makes sense only for content your read from YAML in R. Here, you want to export from knitr rendering to Quarto processing. 

````markdown
---
title: "A title"
---


```{r}
date <- "2024-03-22"
month <- lubridate::month(date)
```

---
date: '`r date`'
---

`r month`

````
![image](https://github.com/quarto-dev/quarto-r/assets/6791940/b5748f29-c813-4b7e-a67a-21f6408520ae)

**Comment by olivroy** [comment]
> **It works today, you just need to drop the !expr which makes sense only for content your read from YAML in R. Here, you want to export from knitr rendering to Quarto processing.

I was sure I tried this, but it didn't work. I will browse quarto discussions / issues and link here if necessary!

Thanks for the explanations
